messaging icon positioning

marcelmoreau opened the pull request:
Our messaging component's icon positioning is not ideal. It does not adapt to changes in font-size and line-height, and will look off on projects that require these variances.

The attached video shows the issues and this PR's fix:
- 0:03 a project w/ smaller font-size, line-height (14px / 1.25) creates icon positioning issue
- 0:10 a project w/ larger font-size, line-height (18px / 1.85) creates icon positioning issue
- 0:12 extra copy shrinks the icon
- 0:19-end – I show how my PR fixes these issues

https://user-images.githubusercontent.com/2116946/110367540-cbfdab80-8015-11eb-8a97-9b902fe6027f.mp4

This introduces a `$base-font-size` and `$base-line-height` variable that are good additions to have in order to promote continuity for other component work in the future. I also removed the width and height attributes from the inline'd SVG code as it's unnecessary – we can control in CSS.

khamer commented:
Awesome, just some minor stuff:

- Let's stick with `$font-stack`, `$font-size`, and `$line-height` - or maybe we should prefix all three with something more meaningful like `$body-`.
- When `$leading` is the same as `$icon-size`, let's add an @if to completely skip the transform property.
- Update messaging--filled.twig with the minor edits you did to messaging.twig.

Thanks @marcelmoreau.

marcelmoreau commented:
@khamer

- I renamed the base variables to be `$body-font-stack`, `$body-font-size`, and `$body-line-height`
- added `@if` conditional for icon's transform
- Updated `messaging--filled.twig`
- also updated messaging's HTML to always be `<div class="messaging__content">` and not `<p class="messaging__content">`

mbfitz reviewed (commented):
All these edits look good - the markup changes are aces and super smart with the font-size/icon/leading rules. Can we also add notes in messaging.scss related to the transform if the leading does not match the icon size? @marcelmoreau
